tests/test_enterprise/api/test_views.py

```diff
@@ -627,20 +614,175 @@ def test_enterprise_customer_courses(
 
         assert response == expected
 
+    @ddt.data(
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, False,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            False, True,
+            '',
+            {
+                'detail': "User must be a staff user or "
+                          "associated with the endpoint's appointed common Enterprise Customer."
+            },
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_DETAIL_ENDPOINT,
+            True, False,  # Is a staff user, so it's allowed.
```

Comment:
If no catalog exists however, then the user *must* be staff to get permission, because there would be nothing to cross-check with. So this test considers that case, and the code below is also thus required.